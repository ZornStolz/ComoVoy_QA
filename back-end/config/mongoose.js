const mongoose = require("mongoose");

const url =
  "mongodb+srv://zorn01:NXtVbSiW67OfOdKQ@icesiadvancedweb.cfp8a.mongodb.net/comoVoy?retryWrites=true&w=majority";

mongoose.connect(url);
const db = mongoose.connection;
db.on("error", console.error.bind(console, "Mongo connection error"));

module.exports = db;
