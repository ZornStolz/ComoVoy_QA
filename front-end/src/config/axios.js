import axios from "axios";

let instance = axios.create({
  baseURL: "localhost:3000/",
  responseType: "json",
  //headers: { "X-Requested-With": "XMLHttpRequest" },
});

export default instance;
