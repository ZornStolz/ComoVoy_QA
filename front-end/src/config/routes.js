import { Route, Routes, BrowserRouter as Router } from "react-router-dom";
import Login from "../pages/Login/Login";
import SignUp from "../pages/SignUp/SignUp";
import Home from "../pages/Home/Home";

const routing = (
  <Router>
    <Routes>
      <Route exact path="/" element={<Login />} />
      <Route exact path="/sign-up" element={<SignUp />} />
      <Route exact path="/home" element={<Home />} />
    </Routes>
  </Router>
);
export default routing;
