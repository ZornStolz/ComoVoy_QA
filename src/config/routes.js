import { Route, Switch, BrowserRouter as Router } from "react-router-dom";
import Home from "../pages/Home/Home";

const routing = (
  <Router>
    <Switch>
      <Route exact path="/home" component={Home} />
      <Route exact path="/login" />
      <Route exact path="/sign-up" />
    </Switch>
  </Router>
);
export default routing;
