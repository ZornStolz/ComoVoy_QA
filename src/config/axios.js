import axios from "axios";

let instance = axios.create({
  baseURL: "https://como-voy-back.herokuapp.com/",
  responseType: "json",
  headers: { "X-Requested-With": "XMLHttpRequest" },
});

export default instance;
