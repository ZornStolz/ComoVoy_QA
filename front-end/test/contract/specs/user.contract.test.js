import { provider } from '../config/init-pact';
import { Matchers } from "@pact-foundation/pact";
import axios from "../../../src/config/axios";

describe('User Service', () => {

    beforeAll(async () => {
        await provider.setup();
        // here we will add the contract
    });

    afterAll(() => provider.finalize());

    describe('When a request to list all users is made', () => {
        beforeEach(async () => {
            await provider.addInteraction({
                uponReceiving: "a request to list all users",
                state: "has users",
                withRequest: {
                    method: "GET",
                    path: "/users/"
                },
                willRespondWith: {
                    status: 200,
                    body: Matchers.eachLike({
                        id: Matchers.like("id"),
                        nombre: Matchers.like("nombre"),
                        contrasena: Matchers.like("contrasena"),
                        correo: Matchers.like("correo@correo.com"),
                        promedioPonderado: Matchers.decimal(),
                        institucion: Matchers.like("institucion")
                    }),
                },
            });
        });

        test('should return the users list', async () => {
            const response = await axios.get("/users/");
            expect(response.data).toMatchSnapshot();
            await provider.verify();
        });

    });

    describe('When a request to list an user is made', () => {
        beforeEach(async () => {
            await provider.addInteraction({
                uponReceiving: "a request to list an user",
                state: "has users",
                withRequest: {
                    method: "GET",
                    path: "/users/id"
                },
                willRespondWith: {
                    status: 200,
                    body: {
                        id: Matchers.like("id"),
                        nombre: Matchers.like("nombre"),
                        contrasena: Matchers.like("contrasena"),
                        correo: Matchers.like("correo@correo.com"),
                        promedioPonderado: Matchers.decimal(),
                        institucion: Matchers.like("institucion")
                    },
                },
            });
        });

        test('should return the asked user', async () => {
            const response = await axios.get("/users/id");
            expect(response.data).toMatchSnapshot();
            await provider.verify();
        });

    });

    describe('When a request to login is made', () => {
        beforeEach(async () => {
            await provider.addInteraction({
                uponReceiving: "a request to login",
                state: "has users",
                withRequest: {
                    method: "POST",
                    path: "/users/login",
                    body: {
                        correo: Matchers.like("correo@correo.com"),
                        contrasena: Matchers.like("contrasena")
                    }
                },
                willRespondWith: {
                    status: 200,
                    body: {
                        id: Matchers.like("id"),
                        nombre: Matchers.like("nombre"),
                        contrasena: Matchers.like("contrasena"),
                        correo: Matchers.like("correo@correo.com"),
                        promedioPonderado: Matchers.decimal(),
                        institucion: Matchers.like("institucion")
                    },
                },
            });
        });

        test('should return an user', async () => {
            const response = await axios.post("/users/login", {
                correo: "correo",
                contrasena: "contrasena"
            });
            expect(response.data).toMatchSnapshot();
            await provider.verify();
        });

    });

    describe('When a request to add an user is made', () => {
        beforeEach(async () => {
            await provider.addInteraction({
                uponReceiving: "a request to add an user",
                state: "has users",
                withRequest: {
                    method: "POST",
                    path: "/users/",
                    body: {
                        nombre: Matchers.like("nombre"),
                        contrasena: Matchers.like("contrasena"),
                        correo: Matchers.like("correo@correo.com"),
                        promedioPonderado: Matchers.decimal(),
                        institucion: Matchers.like("institucion")
                    }
                },
                willRespondWith: {
                    status: 200,
                    body: Matchers.like("Created String")
                },
            });
        });

        test('should notify an user is added', async () => {
            const response = await axios.post("/users/", {
                nombre: "nombre",
                contrasena: "contrasena",
                correo: "correo",
                promedioPonderado: 3.1,
                institucion: "institucion"
            });
            expect(response.data).toMatchSnapshot();
            await provider.verify();
        });

    });

    describe('When a request to update an user is made', () => {
        beforeEach(async () => {
            await provider.addInteraction({
                uponReceiving: "a request to update an user",
                state: "has users",
                withRequest: {
                    method: "PUT",
                    path: "/users/id",
                    body: {
                        id: Matchers.like("id"),
                        nombre: Matchers.like("nombre"),
                        contrasena: Matchers.like("contrasena"),
                        correo: Matchers.like("correo@correo.com"),
                        promedioPonderado: Matchers.decimal(),
                        institucion: Matchers.like("institucion")
                    }
                },
                willRespondWith: {
                    status: 200,
                    body: Matchers.like("Updated String")
                },
            });
        });

        test('should notify an user is updated', async () => {
            const response = await axios.put("/users/id", {
                id: "id",
                nombre: "nombre",
                contrasena: "contrasena",
                correo: "correo",
                promedioPonderado: 3.1,
                institucion: "institucion"
            });
            expect(response.data).toMatchSnapshot();
            await provider.verify();
        });

    });
    describe('When a request to delete an user is made', () => {
        beforeEach(async () => {
            await provider.addInteraction({
                uponReceiving: "a request to delete an user",
                state: "has users",
                withRequest: {
                    method: "DELETE",
                    path: "/users/id",
                },
                willRespondWith: {
                    status: 200,
                    body: Matchers.like("Deleted String")
                },
            });
        });

        test('should notify the user deletion', async () => {
            const response = await axios.delete("/users/id", {
                id: "id",
                nombre: "nombre",
                contrasena: "contrasena",
                correo: "correo",
                promedioPonderado: 3.1,
                institucion: "institucion"
            });
            expect(response.data).toMatchSnapshot();
            await provider.verify();
        });

    });
});
