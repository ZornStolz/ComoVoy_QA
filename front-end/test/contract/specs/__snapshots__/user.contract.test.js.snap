// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Service When a request to add an user is made should notify an user is added 1`] = `"Created String"`;

exports[`User Service When a request to add an user is made should return the correct data 1`] = `"Created String"`;

exports[`User Service When a request to delete an user is made should notify the user deletion 1`] = `"Deleted String"`;

exports[`User Service When a request to list all users is made should return the correct data 1`] = `
Array [
  Object {
    "contrasena": "contrasena",
    "correo": "correo@correo.com",
    "id": "id",
    "institucion": "institucion",
    "nombre": "nombre",
    "promedioPonderado": 13.01,
  },
]
`;

exports[`User Service When a request to list all users is made should return the users list 1`] = `
Array [
  Object {
    "contrasena": "contrasena",
    "correo": "correo@correo.com",
    "id": "id",
    "institucion": "institucion",
    "nombre": "nombre",
    "promedioPonderado": 13.01,
  },
]
`;

exports[`User Service When a request to list an user is made should return the asked user 1`] = `
Object {
  "contrasena": "contrasena",
  "correo": "correo@correo.com",
  "id": "id",
  "institucion": "institucion",
  "nombre": "nombre",
  "promedioPonderado": 13.01,
}
`;

exports[`User Service When a request to list an user is made should return the correct data 1`] = `
Object {
  "contrasena": "contrasena",
  "correo": "correo@correo.com",
  "id": "id",
  "institucion": "institucion",
  "nombre": "nombre",
  "promedioPonderado": 13.01,
}
`;

exports[`User Service When a request to login is made should return an user 1`] = `
Object {
  "contrasena": "contrasena",
  "correo": "correo@correo.com",
  "id": "id",
  "institucion": "institucion",
  "nombre": "nombre",
  "promedioPonderado": 13.01,
}
`;

exports[`User Service When a request to update an user is made should notify an user is updated 1`] = `"Updated String"`;

exports[`User Service When a request to update an user is made should return the correct data 1`] = `"Updated String"`;
